lib/hobbes/lang/preds/dependent.C

```diff
@@ -209,34 +209,34 @@ struct _Record {
 };
 
 struct _Variant {
-  typedef array<std::tuple<array<char>*, Type*, int>>* type;
+  typedef array<tuple<array<char>*, Type*, unsigned int>>* type;
   static MonoTypePtr build(_Variant::type& rec) {
     Variant::Members mems;
     for(unsigned int i = 0; i < rec->size; ++i) {
-      mems.push_back(Variant::Member(makeStdString(std::get<0>(rec->data[i])), typeExprToMonoTypePtr(std::get<1>(rec->data[i])), std::get<2>(rec->data[i])));
+      mems.push_back(Variant::Member(makeStdString(rec->data[i].at<0>()), typeExprToMonoTypePtr(rec->data[i].at<1>()), rec->data[i].at<2>()));
     }
     return Variant::make(mems);
   }
 };
 
 struct _OpaquePtr {
-  typedef std::tuple<array<char>*, int, bool> type;
+  typedef tuple<array<char>*, int, bool> type;
   static MonoTypePtr build(_OpaquePtr::type& op) {
-    return OpaquePtr::make(makeStdString(std::get<0>(op)), std::get<1>(op), std::get<2>(op));
+    return OpaquePtr::make(makeStdString(op.at<0>()), op.at<1>(), op.at<2>());
```

Comment:
For what it's worth, the reason for this difference has to do with memory layout.  An `std::tuple<A,B>` doesn't have the same memory layout as an `std::pair<A,B>`, nor the same memory layout as `struct { A a; B b; }`.  That's the only reason I made the `hobbes::tuple<...>` type, to get a tuple with standard memory layout -- which is necessary so that tuples in hobbes can be mapped into and out of C++ without cost.